### ISSUE TITLE
InvocationTargetException on findr

### ISSUE THREAD

**Issue by jovyntls** [AY2122S1-CS2103-T14-2/tp#255]
my addressbook data to replicate:
`findr soccer` gives InvocationTargetException

```json
{
  "persons" : [ {
    "name" : "Eve",
    "phone" : "234234234",
    "email" : "eve@capt",
    "address" : "CAPT",
    "tagged" : [ ]
  }, {
    "name" : "Alice",
    "phone" : "2134234",
    "email" : "safjk@sadjkfh",
    "address" : "temrbusu",
    "tagged" : [ ]
  }, {
    "name" : "Bob",
    "phone" : "213414",
    "email" : "sdf@hsdfa",
    "address" : "cinna",
    "tagged" : [ "friend" ]
  }, {
    "name" : "Trudy",
    "phone" : "21341",
    "email" : "sdf@sdfjkh",
    "address" : "asdjkfha",
    "tagged" : [ ]
  }, {
    "name" : "newperson",
    "phone" : "1234124",
    "email" : "asdf@kjsadfh",
    "address" : "sadjklfha",
    "tagged" : [ ]
  } ],
  "ccas" : [ {
    "name" : "NUSSO",
    "members" : [ {
      "name" : "Eve",
      "phone" : "234234234",
      "email" : "eve@capt",
      "address" : "CAPT",
      "tagged" : [ ]
    }, {
      "name" : "Bob",
      "phone" : "213414",
      "email" : "sdf@hsdfa",
      "address" : "cinna",
      "tagged" : [ "friend" ]
    } ],
    "reminders" : [ {
      "name" : "weekly music practice",
      "startDate" : "2021-10-31T16:00:00.000+0000",
      "numTimePeriod" : 1,
      "occurrences" : 10,
      "ccaName" : "NUSSO",
      "dates" : [ "2021-10-31T16:00:00.000+0000", "2021-11-30T16:00:00.000+0000", "2021-12-31T16:00:00.000+0000", "2022-01-31T16:00:00.000+0000", "2022-02-28T16:00:00.000+0000", "2022-03-31T16:00:00.000+0000", "2022-04-30T16:00:00.000+0000", "2022-05-31T16:00:00.000+0000", "2022-06-30T16:00:00.000+0000", "2022-07-31T16:00:00.000+0000" ],
      "timePeriod" : "MONTH"
    }, {
      "name" : "practice music weekly",
      "startDate" : "2021-06-09T16:00:00.000+0000",
      "numTimePeriod" : 1,
      "occurrences" : 1,
      "ccaName" : "NUSSO",
      "dates" : [ "2021-06-09T16:00:00.000+0000" ],
      "timePeriod" : "ONE_OFF"
    } ],
    "tagged" : [ "music" ]
  }, {
    "name" : "Soccer",
    "members" : [ {
      "name" : "Bob",
      "phone" : "213414",
      "email" : "sdf@hsdfa",
      "address" : "cinna",
      "tagged" : [ "friend" ]
    }, {
      "name" : "Alice",
      "phone" : "2134234",
      "email" : "safjk@sadjkfh",
      "address" : "temrbusu",
      "tagged" : [ ]
    } ],
    "reminders" : [ {
      "name" : "soccer practice",
      "startDate" : "2021-04-03T16:00:00.000+0000",
      "numTimePeriod" : 1,
      "occurrences" : 17,
      "ccaName" : "Soccer",
      "dates" : [ "2021-04-03T16:00:00.000+0000", "2021-04-10T16:00:00.000+0000", "2021-04-17T16:00:00.000+0000", "2021-04-24T16:00:00.000+0000", "2021-05-01T16:00:00.000+0000", "2021-05-08T16:00:00.000+0000", "2021-05-15T16:00:00.000+0000", "2021-05-22T16:00:00.000+0000", "2021-05-29T16:00:00.000+0000", "2021-06-05T16:00:00.000+0000", "2021-06-12T16:00:00.000+0000", "2021-06-19T16:00:00.000+0000", "2021-06-26T16:00:00.000+0000", "2021-07-03T16:00:00.000+0000", "2021-07-10T16:00:00.000+0000", "2021-07-17T16:00:00.000+0000", "2021-07-24T16:00:00.000+0000" ],
      "timePeriod" : "WEEK"
    } ],
    "tagged" : [ "sports", "active" ]
  }, {
    "name" : "chill soccer",
    "members" : [ {
      "name" : "Alice",
      "phone" : "2134234",
      "email" : "safjk@sadjkfh",
      "address" : "temrbusu",
      "tagged" : [ ]
    }, {
      "name" : "Eve",
      "phone" : "234234234",
      "email" : "eve@capt",
      "address" : "CAPT",
      "tagged" : [ ]
    } ],
    "reminders" : [ {
      "name" : "chill soccer practice",
      "startDate" : "2021-04-05T16:00:00.000+0000",
      "numTimePeriod" : 1,
      "occurrences" : 1,
      "ccaName" : "chill soccer",
      "dates" : [ "2021-04-05T16:00:00.000+0000" ],
      "timePeriod" : "ONE_OFF"
    } ],
    "tagged" : [ ]
  } ],
  "reminders" : [ {
    "name" : "soccer practice",
    "startDate" : "2021-04-03T16:00:00.000+0000",
    "numTimePeriod" : 1,
    "occurrences" : 17,
    "ccaName" : "Soccer",
    "dates" : [ "2021-04-03T16:00:00.000+0000", "2021-04-10T16:00:00.000+0000", "2021-04-17T16:00:00.000+0000", "2021-04-24T16:00:00.000+0000", "2021-05-01T16:00:00.000+0000", "2021-05-08T16:00:00.000+0000", "2021-05-15T16:00:00.000+0000", "2021-05-22T16:00:00.000+0000", "2021-05-29T16:00:00.000+0000", "2021-06-05T16:00:00.000+0000", "2021-06-12T16:00:00.000+0000", "2021-06-19T16:00:00.000+0000", "2021-06-26T16:00:00.000+0000", "2021-07-03T16:00:00.000+0000", "2021-07-10T16:00:00.000+0000", "2021-07-17T16:00:00.000+0000", "2021-07-24T16:00:00.000+0000" ],
    "timePeriod" : "WEEK"
  }, {
    "name" : "chill soccer practice",
    "startDate" : "2021-04-05T16:00:00.000+0000",
    "numTimePeriod" : 1,
    "occurrences" : 1,
    "ccaName" : "chill soccer",
    "dates" : [ "2021-04-05T16:00:00.000+0000" ],
    "timePeriod" : "ONE_OFF"
  }, {
    "name" : "weekly music practice",
    "startDate" : "2021-10-31T16:00:00.000+0000",
    "numTimePeriod" : 1,
    "occurrences" : 10,
    "ccaName" : "NUSSO",
    "dates" : [ "2021-10-31T16:00:00.000+0000", "2021-11-30T16:00:00.000+0000", "2021-12-31T16:00:00.000+0000", "2022-01-31T16:00:00.000+0000", "2022-02-28T16:00:00.000+0000", "2022-03-31T16:00:00.000+0000", "2022-04-30T16:00:00.000+0000", "2022-05-31T16:00:00.000+0000", "2022-06-30T16:00:00.000+0000", "2022-07-31T16:00:00.000+0000" ],
    "timePeriod" : "MONTH"
  } ]
}
```

**Comment by jovyntls** [comment]
also there is a `practice music weekly` reminder here

I think the issue is with snoozing a reminder that has only 1 occurrence left

**Comment by okyntary** [comment]
Closed by #257. I forgot to close this issue when I created the PR; I added a "Close #255" afterwards but it doesn't seem to have closed this issue.
